gravitee-am-repository/gravitee-am-repository-jdbc/src/main/resources/liquibase/changelogs/v4_5_5/4.5.5-scope-approval-index.yml

```diff
@@ -0,0 +1,29 @@
+# Recreate index for scope approval if not exit.
+databaseChangeLog:
+  - changeSet:
+      id: 4.5.5-scope-approval-index
+      author: GraviteeSource Team
+      validCheckSum: ANY
+      onError: MARK_RAN
+      preConditions:
+        - onFail: MARK_RAN
+        - not:
+            - indexExists:
+                tableName: scope_approvals
+                indexName: idx_scope_approvals_domain_client_user_id_extid_scope
+      changes:
+        - createIndex:
+            columns:
+              - column:
+                  name: domain
+              - column:
+                  name: client_id
+              - column:
+                  name: user_source
+              - column:
+                  name: user_external_id
+              - column:
+                  name: scope
+            indexName: idx_scope_approvals_domain_client_user_id_extid_scope
+            tableName: scope_approvals
+            unique: false
```

Comment:
**todo**: we should also add the index on "domain, client_id, user_id, scope". Remember the query is doins a OR operation on `user_id or (src + extId)`

We may also have to remove the index named `idx_scope_approvals_domain_user_id_extid` as split it in two other indexes because in the same way we are searching either by userId or extenalId + src (https://github.com/gravitee-io/gravitee-access-management/pull/5265/files#diff-b9bf6f073973891fce3200a64b3b33e02c22f003e924c1669f85e9260d930e0dR27)

Reply:
as discussed during daily, we can ignore comment on `idx_scope_approvals_domain_user_id_extid ` but we still need to create the index on `domain, client_id, user_id and scope`

Reply:
we have this index: https://github.com/gravitee-io/gravitee-access-management/blob/a276d6a685b3e104346dcc7ecff7926bd9676a9e/gravitee-am-repository/gravitee-am-repository-jdbc/src/main/resources/liquibase/changelogs/v3_5_0/schema.yml#L1851